cherry_picker/cherry_picker.py

```diff
@@ -1,116 +1,295 @@
+            full_sha = get_full_sha_from_short(short_sha)
+            commit_message = self.get_commit_message(short_sha)
+            updated_commit_message = f'[{base}]: {commit_message}. \n(cherry picked from commit {full_sha})'
+            subprocess.check_output(["git", "commit", "-am", updated_commit_message, "--allow-empty"], stderr=subprocess.STDOUT)
+
+            self.push_to_remote(base, cherry_pick_branch)
+
+            self.cleanup_branch(cherry_pick_branch)
+        else:
+            click.echo(u"Refuse to push to master \U0001F61B")
+
+@click.command()
+@click.option('--dry-run', is_flag=True,
+              help="Prints out the commands, but not executed.")
+@click.option('--push', 'pr_remote', metavar='REMOTE',
+              help='git remote to use for PR branches', default='origin')
+@click.option('--abort', 'abort', flag_value=True, default=None,
+              help="Abort current cherry-pick and clean up branch")
+@click.option('--continue', 'abort', flag_value=False, default=None,
+              help="Continue cherry-pick, push, and clean up branch")
+@click.option('--status', 'status', flag_value = True, default=None,
+              help="Get the status of cherry-pick")
+@click.argument('commit_sha1', 'The commit sha1 to be cherry-picked', nargs=1,
+                default = "")
+@click.argument('branches', 'The branches to backport to', nargs=-1)
+def cherry_pick_cli(dry_run, pr_remote, abort, status, commit_sha1, branches):
+
+    click.echo("\U0001F40D \U0001F352 \u26CF")
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    pyconfig_path = os.path.join(current_dir, '/pyconfig.h.in')
+
+    if not os.path.exists(pyconfig_path):
+        os.chdir(os.path.join(current_dir, 'cpython'))
+
     if dry_run:
-        click.echo(f"  dry-run: {cmd}")
-        return True
-    try:
-        subprocess.check_output(cmd.split())
-    except subprocess.CalledProcessError:
-        return False
-    return True
+       click.echo("Dry run requested, listing expected command sequence")
+
+    cherry_picker = CherryPicker(pr_remote, commit_sha1, branches, dry_run=dry_run)
+
+    if abort is not None:
+        if abort:
+            cherry_picker.abort_cherry_pick()
+        else:
+            cherry_picker.continue_cherry_pick()
+
+    elif status:
+        cherry_picker.status()
+    else:
+        cherry_picker.backport()
 
 
-def open_pr(forked_repo, base_branch, cherry_pick_branch, *, dry_run=False):
+def get_base_branch(cherry_pick_branch):
     """
-    construct the url for pull request and open it in the web browser
+    return '2.7' from 'backport-sha-2.7'
     """
-    url = f"https://github.com/python/cpython/compare/{base_branch}...{forked_repo}:{cherry_pick_branch}?expand=1"
-    if dry_run:
-        click.echo(f"  dry-run: Create new PR: {url}")
-        return
-    webbrowser.open_new_tab(url)
+    if '-' in cherry_pick_branch:
```

Comment:
`rpartition` should allow this function to be simplified a bit: https://docs.python.org/3/library/stdtypes.html#str.rpartition

Specifically, I think it's just:

    prefix, sep, base_branch = cherry_pick_branch.rpartition('-')
    return base_branch